### ISSUE TITLE
Embedded structs missing JSON tags

### ISSUE THREAD

**Issue by damemi** [coreos/ignition#917]
We found this in OpenShift back in https://github.com/openshift/machine-config-operator/pull/955 while running Kubebuilder generators, and hit this again recently, so opening an issue to refer to

Several types are missing JSON tags in their embedded fields, for example:
https://github.com/coreos/ignition/blob/master/config/v3_0/types/schema.go#L24-L27
Which throw errors like `encountered struct field "" without JSON tag in type "Directory"`

In the discussion of the above linked PR, it seems that the reason for this is wanting to keep these field names from being seen when serialized. However, this could still be accomplished with JSON tags like:
```go
type Directory struct {
	Node `json:",inline"`
	DirectoryEmbedded1 `json:",inline"`
}
``` 

Would this change be possible? It would fix errors we are seeing, while still serializing these field names as "invisible" in the JSON

**Comment by bgilbert** [comment]
See https://github.com/idubinskiy/schematyper/pull/18.

**Comment by darkmuggle** [comment]
Is this resolved, or still an outstanding issue @damemi ?

**Comment by bgilbert** [comment]
Looks like the upstream schematyper PR is still unmerged.

**Comment by damemi** [comment]
Yeah, the PR to fix the upstream tool causing this issue is still open. Unfortunately it looks like that tool also hasn't been updated since Jan 2019, so it may be slow/abandoned.
